### ISSUE TITLE
`ListBox.SelectAll` sometimes fails to select items when `SelectedItem` is bound

### ISSUE THREAD

**Issue by TomEdwardsEnscape** [AvaloniaUI/Avalonia#13676]
Add this XAML to a window:

``` xaml
<ListBox SelectedItem="{Binding $self.Tag}" SelectionMode="Multiple">
	<ListBoxItem>Hello</ListBoxItem>
	<ListBoxItem>Hello</ListBoxItem>
	<ListBoxItem>Hello</ListBoxItem>
	<ListBoxItem>Hello</ListBoxItem>
	<ListBoxItem>Hello</ListBoxItem>
</ListBox>
```

### 11.0.6
Try to select an item by clicking on it. Nothing happens.

### 11.0.5:
Try to select all items when no items are selected. This can be achieved by tabbing to the `ListBox`, then pressing Ctrl+A. Alternatively, a button with a click event handler can be added. Selected items can be deselected with Ctrl+Click.

- If you try to select all with no items selected, then only the first item will be selected.
- If at least one item is selected, then all items will be selected as expected.

### All versions
The bug is no longer reproducible after removing the `SelectedItem` binding. The example binds to `$self.Tag` for convenience, the source and path of the binding don't matter.

## Expected behavior
- Items can be selected 
- Selecting all items always selects all items.

## Environment 

- OS: Windows or Mac
- Avalonia-Version: 11.0.5 + 11.0.6

**Comment by TomEdwardsEnscape** [comment]
11.0.6 has regressed further, and in the repro case of this bug items can't be selected at all. I've updated the bug to describe the new behaviour.

@grokys presumably #13868 will also fix the 11.0.6 behaviour?

**Comment by timunie** [comment]
@TomEdwardsEnscape does nightly also suffer here in your opinion?
